limitador-server/src/metrics.rs

```diff
@@ -0,0 +1,291 @@
+use std::collections::HashMap;
+use std::ops;
+use std::time::{Duration, Instant};
+use tracing::span::{Attributes, Id};
+use tracing::Subscriber;
+use tracing_subscriber::layer::{Context, Layer};
+use tracing_subscriber::registry::LookupSpan;
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct Timings {
+    idle: u64,
+    busy: u64,
+    last: Instant,
+}
+
+impl Timings {
+    fn new() -> Self {
+        Self {
+            idle: 0,
+            busy: 0,
+            last: Instant::now(),
+        }
+    }
+}
+
+impl ops::Add for Timings {
+    type Output = Self;
+
+    fn add(self, _rhs: Self) -> Self {
+        Self {
+            busy: self.busy + _rhs.busy,
+            idle: self.idle + _rhs.idle,
+            last: if self.last < _rhs.last {
```

Comment:
See [`max`](https://doc.rust-lang.org/std/cmp/trait.Ord.html#method.max) which is implemented for `Instant` if I'm not mistaken